### DISCUSSION TITLE
unable to open image with imagemagick

### DISCUSSION THREAD

**Discussion by bbarbes** [ImageMagick/ImageMagick#3667] [Development]
I'm trying to open a series of images like this example: [image](https://drive.google.com/file/d/1ue2VfK-I5uQjt0PcQ9ZgZY3dHrSPi7eM/view?usp=sharing.)

I know that the image is not corrupted, because I can open it with MatLab. But I want to analyze them with Python, and I am unable to open them

I've tryed using a portable version of ImageMagick, Version: ImageMagick 7.0.11-12 Q16 x64 2021-05-09

When I try opening it, I get an error. 
For example, using
`identify 1.jpg`

I get:
`identify: Unsupported JPEG process: SOF type 0xc3 `1.jpg' @ error/jpeg.c/JPEGErrorHandler/346.`

Any idea? Please, it is important to me, because I need them for quality assurance of a medical device. Tnanks in advance

**Comment by snibgo** [comment]
Microsoft Photos Viewer can't open your 1.jpg "because it appears to be damaged, corrupt or to large".

Different software has different strategies for dealing with corrupt files. ImageMagick is strict.

**Reply by bbarbes** [reply]
Yes, I undestand. But I know they are not corrupted files. They probably have a very extrage format, but the medical device storages the images that way. I tried also ImageJ, irfanview... No way. Only MatLab

**Reply by Jmuccigr** [reply]
I'll affirm what @snibgo says. Rarely I get a file that I can open with one app, but not another, and IM does tend to be stricter. I've had luck re-saving an image from the app that will open it (of course that app was more than a viewer).

**Comment by snibgo** [comment]
I suggest you use MatLab to convert the files to a format that other software can read.

If this is an ongoing problem, with a medical device creating files that common software cannot read, then I suggest you raise the issue with the manufacturer, and with the authors of jpeglib and similar software.

The critical question is: do the medical images comply with standards, or not? If they do comply, then jpeglib etc should adjust to suit. If they don't comply, the manufacturer should adjust.

Possibly, the medical device is old, and complies with standards as they were then, but not with current standards. In that case, it may be difficult to persuade either side to adjust.

EDIT to add: an internet search on "SOF type 0xc3", eg https://crnl.readthedocs.io/jpeg_formats/index.html , suggests this is an "arcane" 16-bit lossless but inefficient format.

**Comment by fmw42** [comment]
What should the image look like?  Mac Preview and Photoshop say it is corrupted. But GraphicsConverter opens it and shows only a gray square on a white background. 

So I would agree with the previous comments. The file may be OK, but follow a very old JPG standard. Many common libraries do not handle old JPG formats. For example, TIFF with old JPG compression via libtiff will not open it either.

**Reply by bbarbes** [reply]
YEEEES. It's exactly "only a gray square on a white background". I didn't know that program, I see that is a payment program. Do you know any other to convert the files? They are many... Thanks a lot

**Reply by fmw42** [reply]
That is an image processing tool for the Mac. I am surprised that opened it. But that is the only other one that I have and use regularly.

**Reply by fmw42** [reply]
Did you try GIMP?

**Reply by bbarbes** [reply]
Yes, I tried GIMP, and also irfanview.
I'll try the hint of snibgo.
Thanks a lot to all of you for your suggestions

**Reply by Jmuccigr** [reply]
> That is an image processing tool for the Mac. I am surprised that opened it. But that is the only other one that I have and use regularly.

I find GC works well. I referred to it in #3206 as opening a tiff that IM wouldn't.

**Comment by snibgo** [comment]
I have searched for open-source code that can read SOF type 0xc3, without luck. If such code is found, perhaps it could be incorporated into IM, perhaps as a special coder.

**Comment by dlemstra** [comment]
Attached image: ![1.jpg](https://user-images.githubusercontent.com/10426229/118141903-28ac8680-b40a-11eb-8837-c04bfcbfe9a2.jpg)

**Comment by dlemstra** [comment]
What happens when you rename it to `1.jls` and then open it with IrfanView. I suspect your file is created in this format: https://en.wikipedia.org/wiki/Lossless_JPEG.

**Reply by bbarbes** [reply]
I tried with 1.jls in IrfanView, ImageJ and ImageMagic. It doesn't work: bad JPEG format or things like that. But thanks a lot
